### ISSUE TITLE
Account for Foot in Kinematics

### ISSUE THREAD

**Issue by Yay295** [XRobots/openDogV3#2]
Based on this diagram from your [kinematics video](https://www.youtube.com/watch?v=eKZIJwJBjEs), the end point of the leg is currently at the tip of the foot. However, as you stated in the video, the tip of the foot is not necessarily the point touching the ground. To account for this the end point of the leg should actually be at the center of the foot when doing the calculations. The point touching the ground would then be one foot's radius from this point.

![New](https://user-images.githubusercontent.com/8922038/146114232-73ce6dd3-97dd-4ab3-8f64-9adecf62df12.jpg)

**Comment by XRobots** [comment]
Yes the legs are too short - in V2 I used the middle of the ball, so the CF tubes should just be a bit longer.  The centre positions will change as the foot rolls over though so there's still some error. Either way the whole thing is compliant so it's negligible.
